torch/csrc/distributed/c10d/reducer.cpp

```diff
@@ -375,6 +393,14 @@ void Reducer::mark_variable_ready(VariableIndex index) {
     torch::autograd::Engine::get_default_engine().queue_callback([=] {
       std::lock_guard<std::mutex> lock(this->mutex_);
```

Comment:
Since we need to unlock below, can we use a `unique_lock` instead, make both unlock explicit?